--- ../../../../bsg_riscv/rocket-chip/riscv-tools/riscv-tests/benchmarks/Makefile.old	2017-01-29 15:21:18.957972322 -0500
+++ ../../../../bsg_riscv/rocket-chip/riscv-tools/riscv-tests/benchmarks/Makefile	2017-01-29 17:59:08.537518417 -0500
@@ -75,7 +75,9 @@
 bmarks_cycles = 80000
 
 %.hex: %
-	elf2hex 16 32768 $< > $@
+	$(eval $@_WIDTH := 16)
+	$(eval $@_LENGTH := `./calc_hex_length.sh $(RISCV_PREFIX)objdump $< $($@_WIDTH)`)
+	elf2hex $($@_WIDTH) $($@_LENGTH) $< > $@
 
 $(bmarks_riscv_dump): %.riscv.dump: %.riscv
 	$(RISCV_OBJDUMP) $< > $@
