--- /work/bits0/ka429/brg-git/bsg_stuff/bsg_riscv/rocket-chip/riscv-tools/riscv-tests/benchmarks/Makefile.old	2017-03-21 12:34:07.827795076 -0400
+++ /work/bits0/ka429/brg-git/bsg_stuff/bsg_riscv/rocket-chip/riscv-tools/riscv-tests/benchmarks/Makefile	2017-03-21 12:35:30.267243553 -0400
@@ -48,8 +48,11 @@
 
 RISCV_PREFIX=riscv64-unknown-elf-
 RISCV_GCC = $(RISCV_PREFIX)gcc
+RISCV_GPP = $(RISCV_PREFIX)g++
 RISCV_GCC_OPTS = -static -std=gnu99 -O2 -ffast-math -fno-common -fno-builtin-printf
+RISCV_GPP_OPTS = -static            -O2 -ffast-math -fno-common -fno-builtin-printf -nostdlib
 RISCV_LINK = $(RISCV_GCC) -T $(bmarkdir)/common/test.ld $(incs)
+RISCV_LINK_CPP = $(RISCV_GPP) -T $(bmarkdir)/common/test.ld $(incs)
 RISCV_LINK_MT = $(RISCV_GCC) -T $(bmarkdir)/common/test-mt.ld
 RISCV_LINK_OPTS = -nostdlib -nostartfiles -ffast-math -lc -lgcc
 RISCV_OBJDUMP = $(RISCV_PREFIX)objdump --disassemble-all --disassemble-zeroes --section=.text --section=.text.startup --section=.data
@@ -75,7 +78,9 @@
 bmarks_cycles = 80000
 
 %.hex: %
-	elf2hex 16 32768 $< > $@
+	$(eval $@_WIDTH := 16)
+	$(eval $@_LENGTH := `./calc_hex_length.sh $(RISCV_PREFIX)objdump $< $($@_WIDTH)`)
+	elf2hex $($@_WIDTH) $($@_LENGTH) $< > $@
 
 $(bmarks_riscv_dump): %.riscv.dump: %.riscv
 	$(RISCV_OBJDUMP) $< > $@
@@ -83,12 +88,16 @@
 $(bmarks_riscv_out): %.riscv.out: %.riscv
 	$(RISCV_SIM) $< > $@
 
+%.o: %.cc
+	$(RISCV_GPP) $(RISCV_GPP_OPTS) $(bmarks_defs) $(bmarks_extra_defs) \
+	             -c $(incs) $< -o $@
+
 %.o: %.c
-	$(RISCV_GCC) $(RISCV_GCC_OPTS) $(bmarks_defs) \
+	$(RISCV_GCC) $(RISCV_GCC_OPTS) $(bmarks_defs) $(bmarks_extra_defs) \
 	             -c $(incs) $< -o $@
 
 %.o: %.S
-	$(RISCV_GCC) $(RISCV_GCC_OPTS) $(bmarks_defs) -D__ASSEMBLY__=1 \
+	$(RISCV_GCC) $(RISCV_GCC_OPTS) $(bmarks_defs) $(bmarks_extra_defs) -D__ASSEMBLY__=1 \
 	             -c $(incs) $< -o $@
 
 riscv: $(bmarks_riscv_dump) $(bmarks_riscv_hex)
